taf/updater/updater.py

```diff
@@ -463,101 +497,105 @@ def _update_named_repository(
+                            library_dir=library_dir,
+                            update_from_filesystem=update_from_filesystem,
+                            expected_repo_type=expected_repo_type,
+                            target_repo_classes=target_repo_classes,
+                            target_factory=target_factory,
+                            only_validate=only_validate,
+                            validate_from_commit=validate_from_commit,
+                            conf_directory_root=conf_directory_root,
+                            visited=visited,
+                            repos_update_data=repos_update_data,
+                            transient_data=transient_data,
+                            out_of_band_authentication=child_auth_repo.out_of_band_authentication,
+                            scripts_root_dir=scripts_root_dir,
+                            checkout=checkout,
+                        )
+                        if error:
+                            raise error
+                    except Exception as e:
+                        errors.append(str(e))
+
+                if len(errors):
+                    errors = "\n".join(errors)
+                    taf_logger.error(
+                        "Update of {} failed. One or more referenced authentication repositories could not be validated:\n {}",
+                        auth_repo.name,
+                        errors,
+                    )
+                    error = UpdateFailedError(
+                        f"Update of {auth_repo.name} failed. One or more referenced authentication repositories could not be validated:\n {errors}"
+                    )
+                    update_status = Event.FAILED
+
+            if (
+                not only_validate
+                and len(commits)
+                and (update_status == Event.CHANGED or update_status == Event.PARTIAL)
+            ):
+                # when performing breadth-first update, validation might fail at some point
+                # but we want to update all repository up to it
+                # so set last validated commit to this last valid commit
+                last_commit = commits[-1]
+                # if there were no errors, merge the last validated authentication repository commit
+                _merge_commit(auth_repo, auth_repo.default_branch, last_commit, True)
+                # update the last validated commit
+                if not excluded_target_globs:
+                    auth_repo.set_last_validated_commit(last_commit)
+
+            # do not call the handlers if only validating the repositories
+            # if a handler fails and we are in the development mode, revert the update
+            # so that it's easy to try again after fixing the handler
+            # JMC: Added "and not no_targets:" to this first line of code
+            if not only_validate and not excluded_target_globs and not no_targets:
```

Comment:
This is correct, but in order to actually avoid update of target repositories and support `no-upstream` flag as well, you'll need to update `updater_pipeline` That's why that issue did not have the "good first issue label".  Handling `no-targets` will probably be pretty easy.  We can skip steps which download and validate target repositories. You'll see that there is an enum called `RunMode` which allows us to configure which steps we'll run depending on if we're running update or local validation. At the moment, we have this:

```
steps=[
                (self.set_existing_repositories, RunMode.UPDATE),
                (self.check_if_local_repositories_clean, RunMode.UPDATE),
....

So, we should somehow extent this so that we can specify that a step should be skipped if `no_targets` is set to `True`. We also have `no_upstream`  which is a bit trickier because we can't know if we should skip update of target repositories until we've executed `clone_or_fetch_users_auth_repo`

An idea that came to my mind now is that we could add another (optional) entry to these tuples, and that would be a method that would be called to check if the step should be executed.

```
(load_target_repositories, , RunMode.ALL, self.check_if_targets_skipped)

def check_if_targets_skipped(self):
  return self.no_targets or (self.no_upstream and self.auth_repo_not_updated)
```
`auth_repo_not_updated` would be set after auth repo validation.
So with those update, updating  this for loop `for step, step_run_mode in self.steps:` should be sufficient